### PR TITLE
feat: Loading Indicator On Captures Page

### PR DESCRIPTION
**Pull request by mitia-anah**
## Description

- Added loading indicator to captures page. While the captures table is loading, the loading indicator is displayed. 

**Issue(s) addressed**

- Resolves #https://github.com/Greenstand/treetracker-admin-client/issues/297

**What kind of change(s) does this PR introduce?**

- [x] Enhancement
- [ ] Bug fix
- [ ] Refactor

**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Issue

**What is the current behavior?**
- It takes several seconds for any records to appear in the Captures table,

**What is the new behavior?**
- The loading indicator is displayed to give the user feedback that loading has not been completed yet.

## Breaking change

**Does this PR introduce a breaking change?**

### CODE REVIEW

**Review by TANguyen1893 — COMMENTED** [review]
Would it be easy to add tests to ensure the changes that are made will not break anything?

**Comment by nmcharlton** [comment]
@mitia-anah Please can you rebase or merge the latest Greenstand master branch into your feature branch?

File: src/components/Captures/CaptureTable.js [hunk 1]
```diff
@@ -258,26 +259,35 @@ const CaptureTable = () => {
...
-                      speciesLookup,
-                      captureTagLookup[capture.id] || [],
-                      attr,
-                      renderer
-                    )}
-                  </TableCell>
-                ))}
-              </TableRow>
-            ))}
+          {captures.length === 0 ? (
```
Comment (nmcharlton) [inline src/components/Captures/CaptureTable.js:262]: Unfortunately, this isn't quite enough. There may be occasions when zero captures match the filters, so we want to show an empty list.
Instead, we will need to add an `isLoading` boolean flag to CapturesContext that tracks whether a Captures load is in progress.

**Review by nmcharlton — CHANGES_REQUESTED** [review]
I've merged in changes from another PR that got lost, so you'll need to do `git pull origin` again before making changes.

File: src/components/Captures/CaptureTable.js [hunk 2]
```diff
@@ -120,7 +121,7 @@ const CaptureTable = () => {
   useEffect(async () => {
     // Don't do anything if there are no captures
     if (!captures?.length) {
-      return;
+      return isLoading;
```
Comment (nmcharlton) [inline src/components/Captures/CaptureTable.js:124]: Nothing uses the return value from this function, so this doesn't do anything.
The `isLoading` flag should be set to `true` before the API is called and `false` when the response is received. See  VerifyContext for guidance:
https://github.com/Greenstand/treetracker-admin-client/blob/4498a21d77b7c02ab1647248df303a603a5da198/src/context/VerifyContext.js#L142

This flag should then be used in CapturesTable to determine whether to show the loading indicator.

**Review by nmcharlton — CHANGES_REQUESTED** [review]

File: src/components/Captures/CaptureTable.js [hunk 3]
```diff
@@ -258,26 +260,35 @@ const CaptureTable = () => {
...
-                      attr,
-                      renderer
-                    )}
-                  </TableCell>
-                ))}
-              </TableRow>
-            ))}
+          {captures.length === 0 ? (
+            <div
+              style={{ position: 'relative', left: '300%', bottom: '-179px' }}
```
Comment (nmcharlton) [inline src/components/Captures/CaptureTable.js:265]: Move the style details to CaptureTable.styles.js as a new class that is referenced with an attribute like `className={classes.loadingIndicator}`.

Comment (nmcharlton) [inline src/components/Captures/CaptureTable.js:265]: This positioning seems odd. Take a look at how `CircularProgress` is positioned in other admin panel tools.

File: src/components/Captures/CaptureTable.js (hunk 1 above)
Comment (nmcharlton) [reply src/components/Captures/CaptureTable.js:263]: This should be `{isLoading ? (`

**Review by nmcharlton — APPROVED** [review]
I've moved things around a little – it's a good idea for the Context to manage the loading state and the Table to just read it.

**Pull request merged**